src/s3/commands/presignedURL.ts

```diff
@@ -3,33 +3,40 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { SignedUrlRequest } from '../../shared/clients/s3Client'
+import * as vscode from 'vscode'
+
+import { DefaultS3Client, SignedUrlRequest } from '../../shared/clients/s3Client'
 import { Env } from '../../shared/vscode/env'
 import { copyToClipboard } from '../../shared/utilities/messages'
 import { S3FileNode } from '../explorer/s3FileNode'
 import { Window } from '../../shared/vscode/window'
-import { localize } from '../../shared/utilities/vsCodeUtils'
 import { invalidNumberWarning } from '../../shared/localizedText'
 import { telemetry } from '../../shared/telemetry/telemetry'
 import { ToolkitError } from '../../shared/errors'
+import { Wizard } from '../../shared/wizards/wizard'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { createCommonButtons } from '../../shared/ui/buttons'
+import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createRegionPrompter } from '../../shared/ui/common/region'
+import { getLogger } from '../../shared/logger'
 
 export async function presignedURLCommand(
-    node: S3FileNode,
+    node?: S3FileNode,
     window = Window.vscode(),
     env = Env.vscode()
 ): Promise<void> {
     await telemetry.s3_copyUrl.run(async span => {
         span.record({ presigned: true })
 
-        const validTime = await promptTime(node.file.key, window)
-        const s3Client = node.s3
-        const request: SignedUrlRequest = {
-            bucketName: node.bucket.name,
-            key: node.file.key,
-            time: validTime * 60,
-            operation: 'getObject',
+        const response = await new PresignedUrlWizard(node).run()
+        if (!response) {
+            getLogger().debug('s3: PresignedUrlWizard returned undefined. User cancelled.')
```

Comment:
it's good to log stuff like this, but I think the wizard framework already do so. Can you confirm?

Reply:
The wizard won't log but the top-level command will if it catches certain exceptions. `CancellationError` didn't exist when the wizard code was written but it should probably be updated to throw instead of returning `undefined`.